languages/csharp/reference/README.md

```diff
@@ -8,89 +8,131 @@ The C# concept exercises are based on concepts. The list below contains the conc
+- Fields
+- Properties
+- Indexers
 
 ### Functional
 
 While C# is primarily an OO language, lots of functional concepts have been added to the language:
 
-- [Immutability](../../../reference/concepts/immutability.md)
-- [Pattern matching](../../../reference/concepts/pattern_matching.md)
-- [Higher-order functions](../../../reference/concepts/higher_order_functions.md)
-- [Type inference](../../../reference/concepts/type_inference.md)
-- [Anonymous methods](../../../reference/concepts/anonymous_functions.md)
-- [Recursion](../../../reference/concepts/recursion.md)
-- [Local functions](../../../reference/concepts/nested_functions.md)
-- [Pipelines (LINQ)](../../../reference/concepts/pipelines.md)
-
-### Platform-specific
-
-- [Assemblies](../../../reference/tooling/dotnet-assemblies.md)
-
-### Memory management
-
-- Resource passing
-- [Resource allocation](./memory_allocation.md) (can include `Span<T>`/`Memory<T>` types)
-- Resource cleanup
-- Resource lifetime
-
-### Methods
-
-- Method overloading
-- Named arguments
-- Optional arguments
-- Extension methods (mixin)
+- [Functions][functions]
+  - [Higher-order functions][higher_order_functions]
+  - [Anonymous functions][anonymous_functions]
+  - [Local functions][local_functions]
+- [Immutability][immutability]
+- [Pattern matching][pattern_matching]
+- [Type inference][type_inference]
+- [Recursion][recursion]
+- [LINQ][linq]
+- Expressions vs statements
 
 ### General
 
-- Arithmetic overflow
-- [Sameness](../../../reference/concepts/sameness.md)
-- [Conditionals](../../../reference/concepts/conditionals.md)
-- [Enumeration](../../../reference/concepts/enumeration.md)
-- Iterators (yield)
-- Namespaces
-- [Generics](../../../reference/concepts/generics.md)
-- Exception handling
-- Implicit/explicit conversion
-- Boxing/unboxing
-- Anonymous types
-- Concurrency (can include concurrent collections)
+- [Variables]
+  - Assignment
+  - Default values (a `bool` being `false` by default, etc.)
+- Scoping
+  - Namespaces
+  - Imports (usings)
+  - Visibility (`public`, `private`, etc.)
+- Numbers
+  - Math operators
+  - Bitwise manipulation
+  - Arithmetic overflow
+- Comparison
+  - [Equality][equality] (`Equals`, `GetHashCode`)
+  - Ordering
+- Conditionals
+  - Boolean logic
+  - [Conditionals: if][conditionals]
+  - [Conditionals: ternary][conditionals]
+  - Conditionals: while
+  - Conditionals: do-while
+  - Conditionals: switch
+- Enumeration
+  - [Enumeration: for loop][enumeration]
+  - [Enumeration: foreach loop][enumeration]
+- Collections
+  - [Generics][generics]
+  - Iterators (yield)
+  - Collections: mapping
+  - Collections: filtering
+  - Collections: reducing
+  - Collections: ordering
+  - Collections: combining
+- Resources
+  - Resource passing (by reference/by value)
+  - [Resource allocation][memory_allocation]
+  - Resource cleanup (`IDisposable`)
+  - Resource lifetime
+- Conversions
+  - Implicit
+  - Explicit (casts)
+  - Boxing/unboxing
+- Null
```

Comment:
I would like to add the [null-conditional operators](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-). Or it could be placed under the operators.